src/delta/delta.c

```diff
@@ -207,102 +208,110 @@ static int found_override(const char *top, const char *bottom) {
         return k;
 }
 
-static int enumerate_dir_d(Hashmap *top, Hashmap *bottom, Hashmap *drops, const char *toppath, const char *drop) {
-        _cleanup_free_ char *unit = NULL;
-        _cleanup_free_ char *path = NULL;
+static int enumerate_dir_d(Hashmap *bottom, const char *toppath, const char *drop) {
```

Comment:
Maybe it's just me, but I find those variable names very confusing. Maybe s/toppath/dirpath/, s/drop/dirname/, s/path/servicepath/, s/pathkey/drops_key/, s/key/bottom_key/ ?